### PR TITLE
Adds E2E test for alerting in Promscale.

### PR DESCRIPTION
**Pull request by Harkishen-Singh**
Signed-off-by: Harkishen-Singh <harkishensingh@hotmail.com>

This commit mimics a case of actual alerting + alertmanager.
It starts the rules-manager with a alert that is always firing.

Then, in E2E test, we start a mock alertmanager server in the `alertsReceiver()`
which listens for alerts just as the alertmanager does. We send down a channel
to this server that signals whether we received an expected alert or not, which
decides the fate of the test.

All references are linked in the commit.

### CODE REVIEW

File: pkg/pgmodel/ingestor/dispatcher.go [hunk 1]
```diff
@@ -45,9 +45,10 @@ type pgxDispatcher struct {
 	copierReadRequestCh    chan<- readRequest
 	seriesEpochRefresh     *time.Ticker
 	doneChannel            chan struct{}
-	doneWG                 sync.WaitGroup
-	labelArrayOID          uint32
 	closed                 *uber_atomic.Bool
+	metricBatchers         sync.WaitGroup
```
Comment (niksajakovljevic) [inline pkg/pgmodel/ingestor/dispatcher.go:49]: I am totally confused here. Can you please explain why do we need to change batcher and disaptcher for alerting e2e tests? I'm quite against this change especially as part of this PR which imho should have nothing to do with how we do batching...

File: pkg/pgmodel/ingestor/metric_batcher.go [hunk 2]
```diff
@@ -139,8 +140,7 @@ func runMetricBatcher(conn pgxconn.PgxConn,
 	completeMetricCreationSignal chan struct{},
 	metricTableNames cache.MetricCache,
 	copierReadRequestCh chan<- readRequest,
-	labelArrayOID uint32) {
```
Comment (niksajakovljevic) [inline pkg/pgmodel/ingestor/metric_batcher.go:142]: Ohh I see this has never been used. Good catch!

**Review by niksajakovljevic — APPROVED** [review]

File: go.mod [hunk 3]
```diff
@@ -58,6 +58,8 @@ require (
 	gopkg.in/yaml.v2 v2.4.0
 )
 
+require github.com/prometheus/alertmanager v0.24.0
```
Comment (antekresic) [inline go.mod:61]: Why is this not grouped together with above?

File: pkg/tests/end_to_end_tests/alerts_test.go [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+	conf, err := amConfig.LoadFile(amConfigPath)
+	require.NoError(t, err)
+	apiAM.Update(conf, func(model.LabelSet) {})
+
+	u, err := url.Parse("http://localhost:8080")
+	require.NoError(t, err)
+
+	router := route.New()
+	mux := apiAM.Register(router, u.Path)
+	srv := &http.Server{Addr: ":8080", Handler: mux}
```
Comment (antekresic) [inline pkg/tests/end_to_end_tests/alerts_test.go:153]: Lets use httptest server here if possible.

**Review by antekresic — CHANGES_REQUESTED** [review]
Lets try updating the test to use httptest server and try to reduce the amount of synchronization there (I feel like it can be simplified quite a bit).

File: pkg/pgmodel/ingestor/dispatcher.go [hunk 5]
```diff
@@ -230,7 +231,6 @@ func (p *pgxDispatcher) InsertTs(ctx context.Context, dataTS model.Data) (uint64
 				maxt = ts
 			}
 		}
-		// the following is usually non-blocking, just a channel insert
```
Comment (antekresic) [inline pkg/pgmodel/ingestor/dispatcher.go:233]: Why is this removed?

**Review by antekresic — APPROVED** [review]
Approved with a question.

**Pull request merged**